src/app/lab/lab.component.ts

```diff
@@ -26,26 +26,43 @@ import * as _ from 'lodash';
   templateUrl: './lab.component.html'
 })
 export class LabComponent implements OnInit, OnDestroy {
-  name = 'Climate Lab';
 
-  constructor(viewContainerRef: ViewContainerRef,
-              private chartService: ChartService,
+  public apiCities: string = apiHost + "/api/city/?search=:keyword&format=json";
+  public climateModels: ClimateModel[] = [];
+  public scenarios: Scenario[] = [];
+  public project: Project;
+
+  constructor(private chartService: ChartService,
               private climateModelService: ClimateModelService,
               private scenarioService: ScenarioService,
               private projectService: ProjectService) {
-    // necessary to catch application root view container ref. see:
-    // https://valor-software.com/ng2-bootstrap/#/modals
-    this.viewContainerRef = viewContainerRef;
```

Comment:
Moved because it was in the wrong place, it needs to be in the constructor for the top-level component, which is AppComponent, not LabComponent.